bears/vcs/git/CISkipInspectBear.py

```diff
@@ -0,0 +1,74 @@
+import re
+
+from bears.vcs.git.GitCommitMetadataBear import GitCommitMetadataBear
+from coalib.bears.GlobalBear import GlobalBear
+from coalib.results.Result import Result
+
+
+class CISkipInspectBear(GlobalBear):
+    LANGUAGES = {'Git'}
+    AUTHORS = {'The coala developers'}
+    AUTHORS_EMAILS = {'coala-devel@googlegroups.com'}
+    LICENSE = 'AGPL-3.0'
+
+    CI_PROVIDERS_METADATA = {
+          'CircleCI': r'\[ci skip\]|\[skip ci\]',
+          'TravisCI': r'\[ci skip\]|\[skip ci\]',
+          'AppveyorCI': r'\[skip ci\]|\[ci skip\]|\[skip appveyor\]',
+          }
+
+    BEAR_DEPS = {GitCommitMetadataBear}
+
+    def run(self, dependency_results,
+            ci_engines_used: list = [],
+            filename_globs: list = [],
+            **kwargs):
+        """
+        Inspect the HEAD commit to check if it disables
+        CI build.
+
+        :param ci_engines_used:     CI engines used in
+                                    the project.
+        :param filename_globs:      To identify files
+                                    that should not skip
+                                    CI builds.
+        """
+        for result in dependency_results[GitCommitMetadataBear.name]:
+
+            for ci_engine in ci_engines_used:
+
+                if ci_engine not in self.CI_PROVIDERS_METADATA:
+                    raise RuntimeError('%s is not supported by '
+                                       'CISkipInspectBear.' % (ci_engine))
+
+                supported_regex = self.CI_PROVIDERS_METADATA[ci_engine]
+
+                # AppVeyor allows to skip build by commit if the
+                # supported sequence is present in the title of the
+                # commit only. All other CI tools allow these sequences
+                # to be present anywhere in the commit message.
+                if ci_engine == 'AppveyorCI':
+                    pos = result.raw_commit_message.find('\n')
+                    commit_title = result.raw_commit_message[:pos] if (
+                        pos != -1) else result.raw_commit_message
+
+                    match = re.search(supported_regex, commit_title)
+
+                else:
+                    match = re.search(
+                        supported_regex, result.raw_commit_message)
+
+                if match:
+
+                    all_files = (result.modified_files +
+                                 result.added_files + result.deleted_files)
+
+                    for file in all_files:
+                        for glob in filename_globs:
```

Comment:
if there are no globs, this will do nothing, and report no errors.

Use RuntimeError or something if the user hasnt supplied necessary settings.